edgelet/edgelet-http/src/util/incoming.rs

```diff
@@ -70,45 +71,50 @@ impl Stream for Incoming {
                     .lock()
                     .expect("Unable to lock the connections mutex");
 
-                // Look through the connections list for the first connection that is either ready to be
-                // passed to the stream selector or has failed and needs the error bubbled up.
-                // Return a tuple containing the index and state.
-                let val = connections
-                    .iter_mut()
-                    .map(|(fut, _)| fut.poll())
-                    .enumerate()
-                    .find(|(_, result)| match result {
-                        Ok(v) => v.is_ready(),
-                        Err(_) => true,
-                    });
+                loop {
+                    // Look through the connections list for the first connection that is either ready to be
+                    // passed to the stream selector or has failed and needs the error bubbled up.
+                    // Return a tuple containing the index and state.
+                    let val = connections
+                        .iter_mut()
+                        .map(|(fut, _)| fut.poll())
+                        .enumerate()
+                        .find(|(_, result)| match result {
+                            Ok(v) => v.is_ready(),
+                            Err(_) => true,
+                        });
 
-                // Validate that the poll is ready, and remove that value. Then return the connection.
-                // If no connections are available in connection manager, return Async::NotReady
-                match val {
-                    Some((i, result)) => {
-                        let (_, addr) = connections.remove(i);
-                        match result {
-                            Ok(Async::Ready(tls_stream)) => {
-                                #[cfg(not(windows))]
-                                {
-                                    Async::Ready(Some((StreamSelector::Tls(tls_stream), addr)))
+                    // Validate that the poll is ready, and remove that value. Then return the connection.
+                    // If no connections are available in connection manager, return Async::NotReady
+                    match val {
+                        Some((i, result)) => {
+                            let (_, addr) = connections.remove(i);
+                            match result {
+                                Ok(Async::Ready(tls_stream)) => {
+                                    #[cfg(not(windows))]
+                                    {
+                                        return Ok(Async::Ready(Some((
+                                            StreamSelector::Tls(tls_stream),
+                                            addr,
+                                        ))));
+                                    }
+                                    #[cfg(windows)]
```

Comment:
@kkdawkins Why do we need this block for windows? I thought whole TLS thing is for [unix only](https://github.com/Azure/iotedge/blob/a11c02fb733c576853b609c387a81faeab11ab0b/edgelet/edgelet-http/src/util/incoming.rs#L55)